hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/meta/CkpMetadata.java

```diff
@@ -96,9 +97,11 @@ public void close() {
    *
    * <p>This expects to be called by the driver.
    */
-  public void bootstrap() throws IOException {
+  public void bootstrap(HoodieTableMetaClient metaClient) throws IOException {
     fs.delete(path, true);
     fs.mkdirs(path);
+    TimelineUtils.getBasicCommitTimeline(metaClient).filterPendingExcludingCompaction()
```

Comment:
if there are multi flink jobs writing to hudi, the logic here would not solve the problem that one flink job may use other flink job pending instant. However, it would be solved in following PR.

Reply:
Yeah, we should use different `CkpMetadata` for different jobs, and we may not clean the metadata at all. Or a better solution is just to keep the old `CkpMetadata` messages. There is no need to clean it actually.